# MỤC TIÊU

Thực hiện bài toán kiểm định 1 phía cho 2 tham số

- Kỳ vọng ($\mu$)
- Tỷ lệ (p)

# VÍ DỤ

## Ví dụ 1

![image.png](attachment:a05ecccb-fe85-42f9-9ddb-d2e30684cefb.png)

In [ ]:
import numpy as np
import scipy.stats as st

In [ ]:
freq_table = [(155, 3), (165, 9), (175, 11), (185, 3), (195, 2), (205, 1)]
# freq_table = [(155, 3), (165, 2), (175, 1)]

In [ ]:
data = []
for v, f in freq_table:
    for i in range(f):
        data.append(v)

# data

In [ ]:
# Tinh toan cac so lieu thong ke tren mau

# Kich thuoc mau
n = len(data)

# Tinh trung binh mau
tb = round(np.mean(data), 4)

# Tinh do lech chuan mau
dlc = round(np.std(data, ddof=1), 4)

print('Kich thuoc mau: ', n)
print('Trung binh mau: ', tb)
print('Do lech chuan mau: ', dlc)

Kich thuoc mau:  29
Trung binh mau:  173.2759
Do lech chuan mau:  11.9729


In [ ]:
import math

In [ ]:
# Cau b: Tim khoang uoc luong cho ky vong o do tin cay 95%

st.t.interval(0.95, df=n-1, loc=tb, scale=(dlc/math.sqrt(n)))

(168.72165240566306, 177.83014759433695)

In [ ]:
# Cau b (tinh tay)

# Vì ta không biết giá trị của sigma (độ lệch chuẩn quần thể), 
# vì vậy, trung bình mẫu tuân theo phân phối Student
# Ta có: sai số ước lượng:
epsilon = st.t.isf(0.025, df=n-1) * (dlc / math.sqrt(n))

epsilon

4.554238369648043

In [ ]:
# Khoảng ước lượng:
print('Khoảng ước lượng: ({0}, {1})'.format(tb-epsilon, tb+epsilon))

Khoảng ước lượng: (168.72166163035197, 177.83013836964804)


__Câu c: Kiểm định__

Phát biểu: "Lượng cholesterol trung bình là 175mg%".

B1: Đặt cập giả thuyết, đối thuyết

Ho = $\mu = 175$

Ha = $\mu <> 175$

Giả định Ho đúng, tìm bằng chứng để thử bác bỏ Ho

B2: Mức ý nghĩa của kiểm định: $\alpha$ = 0.05

B4: Xác định miền bác bỏ giả thuyết:        

In [ ]:
t_alpha_chia_2 = st.t.isf(0.025, df=n-1)
t_alpha_chia_2

2.048407141795244

Vậy miền bác bỏ của giả thuyết là: {t: |t| > 2.0484}}

In [ ]:
# B3: Tính trị kiểm định:
        
t = (tb - 175) / (dlc / math.sqrt(n))
t

-0.7754648116981316

B5: Kết luận

t không thuộc miền bác bỏ.

Vậy chấp nhận giả thuyết Ho ban đầu

## Ví dụ 2

# BÀI TẬP

## Kiểm định đặc trưng tỉ lệ (p)

### Bài 1

Dataset 18 (__M&M__) chứa thông tin trọng lượng và màu sắc của mẫu các viên kẹo của hãng M&M. 

Sử dụng mẫu trên bạn hãy kiểm tra phát biểu sau: __"20% kẹo M&M có màu đỏ"__ với mức ý nghĩa là 5%

In [36]:
import numpy as np
import scipy.stats as st
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
df = pd.read_excel('/content/drive/My Drive/Colab Notebooks/18_M_M.xls')
df

Red  Orange  Yellow  Brown   Blue  Green
0   0.751   0.735   0.883  0.696  0.881  0.925
1   0.841   0.895   0.769  0.876  0.863  0.914
2   0.856   0.865   0.859  0.855  0.775  0.881
3   0.799   0.864   0.784  0.806  0.854  0.865
4   0.966   0.852   0.824  0.840  0.810  0.865
5   0.859   0.866   0.858  0.868  0.858  1.015
6   0.857   0.859   0.848  0.859  0.818  0.876
7   0.942   0.838   0.851  0.982  0.868  0.809
8   0.873   0.863     NaN    NaN  0.803  0.865
9   0.809   0.888     NaN    NaN  0.932  0.848
10  0.890   0.925     NaN    NaN  0.842  0.940
11  0.878   0.793     NaN    NaN  0.832  0.833
12  0.905   0.977     NaN    NaN  0.807  0.845
13    NaN   0.850     NaN    NaN  0.841  0.852
14    NaN   0.830     NaN    NaN  0.932  0.778
15    NaN   0.856     NaN    NaN  0.833  0.814
16    NaN   0.842     NaN    NaN  0.881  0.791
17    NaN   0.778     NaN    NaN  0.818  0.810
18    NaN   0.786     NaN    NaN  0.864  0.881
19    NaN   0.853     NaN    NaN  0.825    NaN
20    NaN   0.864     NaN    NaN  0.855    NaN
21    NaN   0.873     NaN    NaN  0.942    NaN
22    NaN   0.880     NaN    NaN  0.825    NaN
23    NaN   0.882     NaN    NaN  0.869    NaN
24    NaN   0.931     NaN    NaN  0.912    NaN
25    NaN     NaN     NaN    NaN  0.887    NaN
26    NaN     NaN     NaN    NaN  0.886    NaN

In [ ]:
# Tinh toan cac so lieu thong ke tren mau

# Kich thuoc mau
n = df.count().sum()

#Tính muy
muy = n *20 /100

# Tinh trung binh mau
tb = round(df.count().mean(), 4)

# Tinh do lech chuan mau
dlc = round(df.count().std(), 4)

print('Kich thuoc mau: ', n)
print('Muy: ', muy)
print('Trung binh mau: ', tb)
print('Do lech chuan mau: ', dlc)

Kich thuoc mau:  100
Muy:  20.0
Trung binh mau:  16.6667
Do lech chuan mau:  8.3106


In [ ]:
#Tính miền bác bỏ
t_alpha_chia_2 = st.t.isf(0.025, df=n-1)
t_alpha_chia_2
print("Miền bác bỏ của giả thuyết là:{t:|t| >", t_alpha_chia_2,"}")

Miền bác bỏ của giả thuyết là:{t:|t| > 1.9842169515086832 }


In [ ]:
#Tính trị số kiểm định 
import math
t = (tb - muy) / (dlc / math.sqrt(n))
print("Trị số kiểm định: ",t)

Trị số kiểm định:  -4.010901739946575


Vậy trị số kiểm định nằm trong vùng bác bỏ => bác bỏ giả thuyết "20% kẹo M&M là màu đỏ"

### Bài 2

Dataset 3 (__FRESH15__) chứa thông tin khảo sát những thay đổi về trọng lượng (weight - WT) và chỉ số khối của cơ thể (BMI) của sinh viên năm nhất tại một trường Đại Học. Thời điểm khảo sát là bắt đầu học kỳ (tháng 9 - SEP) và kết thúc học kỳ (tháng 4 - APR), các sinh viên được chọn ngẫu nhiên trong trường.

Sử dụng mẫu trên, hãy kiểm tra phát biểu:__"50% số sinh viên trong trường là nam"__ với mức ý nghĩa 50%

In [ ]:
df2 = pd.read_excel('/content/drive/My Drive/Colab Notebooks/03_FRESH15.xls')
df2

SEX  WTSEP  WTAPR  BMISP  BMIAP
0    M     72     59  22.02  18.14
1    M     97     86  19.70  17.44
2    M     74     69  24.09  22.43
3    M     93     88  26.97  25.57
4    F     68     64  21.51  20.10
..  ..    ...    ...    ...    ...
62   M     65     71  22.51  24.45
63   M     75     82  23.69  25.80
64   F     42     49  15.08  17.74
65   M     74     82  22.64  25.33
66   M     94    105  36.57  40.86

[67 rows x 5 columns]

In [ ]:
df2_Sex = df2['SEX']
 
# Kich thuoc mau
n = len(df2_Sex)

#Tính muy
muy = n *50 /100

# Tinh trung binh mau
tb = round(df2_Sex.value_counts().mean(), 4)

# Tinh do lech chuan mau
dlc = round(df2_Sex.value_counts().std(), 4)

print('Kich thuoc mau: ', n)
print('Muy: ', muy)
print('Trung binh mau: ', tb)
print('Do lech chuan mau: ', dlc)

Kich thuoc mau:  67
Muy:  33.5
Trung binh mau:  33.5
Do lech chuan mau:  2.1213


In [ ]:
#Tính miền bác bỏ
t_alpha_chia_2 = st.t.isf(0.25, df=n-1)
t_alpha_chia_2
print("Miền bác bỏ của giả thuyết là:{t:|t| >", t_alpha_chia_2,"}")

Miền bác bỏ của giả thuyết là:{t:|t| > 0.6782251752793526 }


In [ ]:
#Tính trị số kiểm định 
import math
t = (tb - muy) / (dlc / math.sqrt(n))
print("Trị số kiểm định: ",t)

Trị số kiểm định:  0.0


Vậy trị số kiểm định nằm ngoài vùng bác bỏ => chấp nhận giả thuyết "50% số sinh viên trong trường là nam"

### Bài 3

Dataset 6(__BEARS__) chứa thông tin khảo sát ngẫu nhiên các con gấu trong một khu rừng.

Sử dụng mẫu trên, bạn hãy kiểm tra phát biểu: __"tỷ lệ gấu đực trong khu rừng là 50%""__ với mức ý nghĩa 90%.

Biết rằng thuộc tính SEX có hai giá trị: 1 - gấu đực, 2- gấu cái.

In [ ]:
df3 = pd.read_excel('/content/drive/My Drive/Colab Notebooks/06_BEARS.xls')
df3

AGE  MONTH  SEX  HEADLEN  HEADWTH  NECK  LENGTH  CHEST  WEIGHT
0    19      7    1     11.0      5.5  16.0    53.0   26.0      80
1    55      7    1     16.5      9.0  28.0    67.5   45.0     344
2    81      9    1     15.5      8.0  31.0    72.0   54.0     416
3   115      7    1     17.0     10.0  31.5    72.0   49.0     348
4   104      8    2     15.5      6.5  22.0    62.0   35.0     166
5   100      4    2     13.0      7.0  21.0    70.0   41.0     220
6    56      7    1     15.0      7.5  26.5    73.5   41.0     262
7    51      4    1     13.5      8.0  27.0    68.5   49.0     360
8    57      9    2     13.5      7.0  20.0    64.0   38.0     204
9    53      5    2     12.5      6.0  18.0    58.0   31.0     144
10   68      8    1     16.0      9.0  29.0    73.0   44.0     332
11    8      8    1      9.0      4.5  13.0    37.0   19.0      34
12   44      8    2     12.5      4.5  10.5    63.0   32.0     140
13   32      8    1     14.0      5.0  21.5    67.0   37.0     180
14   20      8    2     11.5      5.0  17.5    52.0   29.0     105
15   32      8    1     13.0      8.0  21.5    59.0   33.0     166
16   45      9    1     13.5      7.0  24.0    64.0   39.0     204
17    9      9    2      9.0      4.5  12.0    36.0   19.0      26
18   21      9    1     13.0      6.0  19.0    59.0   30.0     120
19  177      9    1     16.0      9.5  30.0    72.0   48.0     436
20   57      9    2     12.5      5.0  19.0    57.5   32.0     125
21   81      9    2     13.0      5.0  20.0    61.0   33.0     132
22   21      9    1     13.0      5.0  17.0    54.0   28.0      90
23    9      9    1     10.0      4.0  13.0    40.0   23.0      40
24   45      9    1     16.0      6.0  24.0    63.0   42.0     220
25    9      9    1     10.0      4.0  13.5    43.0   23.0      46
26   33      9    1     13.5      6.0  22.0    66.5   34.0     154
27   57      9    2     13.0      5.5  17.5    60.5   31.0     116
28   45      9    2     13.0      6.5  21.0    60.0   34.5     182
29   21      9    1     14.5      5.5  20.0    61.0   34.0     150
30   10     10    1      9.5      4.5  16.0    40.0   26.0      65
31   82     10    2     13.5      6.5  28.0    64.0   48.0     356
32   70     10    2     14.5      6.5  26.0    65.0   48.0     316
33   10     10    1     11.0      5.0  17.0    49.0   29.0      94
34   10     10    1     11.5      5.0  17.0    47.0   29.5      86
35   34     10    1     13.0      7.0  21.0    59.0   35.0     150
36   34     10    1     16.5      6.5  27.0    72.0   44.5     270
37   34     10    1     14.0      5.5  24.0    65.0   39.0     202
38   58     10    2     13.5      6.5  21.5    63.0   40.0     202
39   58     10    1     15.5      7.0  28.0    70.5   50.0     365
40   11     11    1     11.5      6.0  16.5    48.0   31.0      79
41   23     11    1     12.0      6.5  19.0    50.0   38.0     148
42   70     10    1     15.5      7.0  28.0    76.5   55.0     446
43   11     11    2      9.0      5.0  15.0    46.0   27.0      62
44   83     11    2     14.5      7.0  23.0    61.5   44.0     236
45   35     11    1     13.5      8.5  23.0    63.5   44.0     212
46   16      4    1     10.0      4.0  15.5    48.0   26.0      60
47   16      4    1     10.0      5.0  15.0    41.0   26.0      64
48   17      5    1     11.5      5.0  17.0    53.0   30.5     114
49   17      5    2     11.5      5.0  15.0    52.5   28.0      76
50   17      5    2     11.0      4.5  13.0    46.0   23.0      48
51    8      8    2     10.0      4.5  10.0    43.5   24.0      29
52   83     11    1     15.5      8.0  30.5    75.0   54.0     514
53   18      6    1     12.5      8.5  18.0    57.3   32.8     140

In [ ]:
df3_Sex = df3['SEX']

# Kich thuoc mau
n = len(df3_Sex)

#Tính muy
muy = n *50 /100

# Tinh trung binh mau
tb = round(df3_Sex.value_counts().mean(), 4)

# Tinh do lech chuan mau
dlc = round(df3_Sex.value_counts().std(), 4)

print('Kich thuoc mau: ', n)
print('Muy: ', muy)
print('Trung binh mau: ', tb)
print('Do lech chuan mau: ', dlc)

Kich thuoc mau:  54
Muy:  27.0
Trung binh mau:  27.0
Do lech chuan mau:  11.3137


In [ ]:
#Tính miền bác bỏ
t_alpha_chia_2 = st.t.isf(0.81, df=n-1)
t_alpha_chia_2
print("Miền bác bỏ của giả thuyết là:{t:|t| >", t_alpha_chia_2,"}")

Miền bác bỏ của giả thuyết là:{t:|t| > -0.885288551060788 }


In [ ]:
#Tính trị số kiểm định 
import math
t = (tb - muy) / (dlc / math.sqrt(n))
print("Trị số kiểm định: ",t)

Trị số kiểm định:  0.0


Vậy trị số kiểm định nằm trong vùng bác bỏ => bác bỏ giả thuyết "tỷ lệ gấu đực trong khu rừng là 50%"

### Bài 4

Dataset 9 __(MOVIES)__ chứa thông tin khảo sát một số phim trong khoảng thời gian 33 năm gần đây.

Có ý kiến cho rằng: __"55% các phim trong khoảng thời gian 33 năm gần đây có nhãn R (là phim giới hạn độ tuổi - Restricted)"__ .Theo bạn với mức ý nghĩa 95% ý kiến trên có đúng không?

Tham khảo: [Hệ thống phân loại của hiệp hội điện ảnh Mỹ](https://vi.wikipedia.org/wiki/H%E1%BB%87_th%E1%BB%91ng_ph%C3%A2n_lo%E1%BA%A1i_phim_c%E1%BB%A7a_MPAA#Ph%C3%A2n_lo%E1%BA%A1i_v%E1%BB%9Bi_c%C3%A1c_trailer_phim)

In [ ]:
df4 = pd.read_excel('/content/drive/My Drive/Colab Notebooks/09_MOVIES.xls')
df4

Title   MPAA  Budget  Gross  Length  \
0                                  8 Mile      R    41.0    117     110   
1                       Alone in the Dark      R    20.0      5      96   
2                                 Aviator  PG-13   116.0    103     170   
3                                Big Fish  PG-13    70.0     66     125   
4                         Bourne Identity  PG-13    75.0    121     119   
5                                Break-Up  PG-13    52.0    116     105   
6         Charlie's Angels: Full Throttle  PG-13   120.0    101     106   
7                              Collateral      R    65.0    100     120   
8                                   Crash      R     6.5     55     113   
9                          Daddy Day Care     PG    60.0    104      92   
10                           DaVinci Code  PG-13   125.0    213     149   
11  Eternal Sunshine of the Spotless Mind      R    20.0     34     108   
12                   From Justin to Kelly     PG     5.0     12      81   
13            Harry Potter Goblet of Fire  PG-13   150.0    290     157   
14                                 Hostel      R     4.5     47      94   
15                      House of the Dead      R     7.0     10      90   
16                           Last Samurai      R   100.0    111     154   
17                    Million Dollar Baby  PG-13    30.0    100     132   
18         Pirates of the Carribbean (II)  PG-13   225.0    322     150   
19                             Rollerball  PG-13    70.0     19      97   
20                               S.W.A.T.  PG-13    80.0    117     117   
21                          Secret Window  PG-13    40.0     48      96   
22                                  Signs  PG-13    70.0    228     106   
23                            Silent Hill      R    50.0     47     127   
24                        Son of the Mask     PG    74.0     17      94   
25                           Spider-Man 2  PG-13   200.0    373     127   
26                          Star Wars III  PG-13   113.0    380     140   
27                       Sum of All Fears  PG-13    68.0    118     124   
28                            The Pianist      R    35.0     33     150   
29                            The Village  PG-13    72.0    114     108   
30                            Van Helsing  PG-13   160.0    120     132   
31                            Vanilla Sky      R    68.0    101     136   
32                          Walk the Line  PG-13    29.0    120     136   
33                      War of the Worlds  PG-13   132.0    234     116   
34                       Wedding Crashers      R    40.0    209     119   

    Rating  
0      6.7  
1      2.2  
2      7.6  
3      8.0  
4      7.4  
5      5.8  
6      4.8  
7      7.7  
8      8.3  
9      5.7  
10     6.5  
11     8.6  
12     1.9  
13     7.8  
14     5.8  
15     2.0  
16     7.8  
17     8.4  
18     7.5  
19     2.7  
20     6.0  
21     6.3  
22     7.0  
23     6.6  
24     2.0  
25     7.8  
26     8.0  
27     6.4  
28     8.5  
29     6.6  
30     5.3  
31     6.8  
32     8.1  
33     6.7  
34     7.3

In [ ]:
df4_MPAA = df4['MPAA']

# Kich thuoc mau
n = len(df4_MPAA)

#Tính muy
muy = n *55 /100

# Tinh trung binh mau
tb = round(df4_MPAA.value_counts().mean(), 4)

# Tinh do lech chuan mau
dlc = round(df4_MPAA.value_counts().std(), 4)

print('Kich thuoc mau: ', n)
print('Muy: ', muy)
print('Trung binh mau: ', tb)
print('Do lech chuan mau: ', dlc)

Kich thuoc mau:  35
Muy:  19.25
Trung binh mau:  11.6667
Do lech chuan mau:  8.5049


In [ ]:
#Tính miền bác bỏ
t_alpha_chia_2 = st.t.isf(0.95**2, df=n-1)
t_alpha_chia_2
print("Miền bác bỏ của giả thuyết là:{t:|t| >", t_alpha_chia_2,"}")

Miền bác bỏ của giả thuyết là:{t:|t| > -1.3219813831584992 }


In [ ]:
#Tính trị số kiểm định 
import math
t = (tb - muy) / (dlc / math.sqrt(n))
print("Trị số kiểm định: ",t)

Trị số kiểm định:  -5.275007092285543


Vậy trị số kiểm định nằm trong vùng bác bỏ => bác bỏ giả thuyết "55% các phim trong khoảng thời gian 33 năm gần đây có nhãn R (là phim giới hạn độ tuổi - Restricted)"

## Kiểm định đặc trưng kỳ vọng ($\mu$) biết ($\sigma$)

### Bài 5

Dataset 19(__SCREWS__) chứa thông tin khảo sát chiều dài một số đinh ốc được cung cấp bởi công ty Crown Bolt (đơn vị inches)

Trên bao bì của các đinh ốc do công ty sản xuất có ghi chiều dài của các đinh ốc là 3/4 inches. Sử dụng mẫu trên bạn hãy cho biết chiều dài của đinh ốc lấy từ mẫu thu nhận được từ dữ liệu trên có nhất quán với chiều dài ghi trên bao bì hay không? Kiểm định với mức ý nghĩa 5%

_Biết rằng, theo tài liệu kỹ thuật của công ty thì độ lệch chuẩn về chiều dài các con vít là 0.012 inches_

In [41]:
df5 = pd.read_excel('/content/drive/My Drive/Colab Notebooks/19_SCREWS.xls')
df5

FileNotFoundError: ignored

### Bài 6

Dataset 13 __(VOLTAGE)__ liệt kê các mức điện áp được đo tại các hộ gia đình bởi các máy đo khác nhau: tại nhà (HOME), tại máy phát độc lập (Generator), tại trạm phát (UPS).

Công ty cung cấp điện cho vùng này cam kết rằng: __"Mục tiêu cung cấp điện cho mỗi hộ gia đình là 120V"__ . Biết rằng độ lệch chuẩn của tất cả các mức điện áp đo được là 0.24V.

Với mức ý nghĩa 1%. Bạn hãy kiểm tra xem công ty có làm đúng cam kết hay không?

In [42]:
df6 = pd.read_excel('/content/drive/My Drive/Colab Notebooks/13_VOLTAGE.xls')
df6

Day   Home  Generator    UPS
0     1  123.8      124.8  123.1
1     2  123.9      124.3  123.1
2     3  123.9      125.2  123.6
3     4  123.3      124.5  123.6
4     5  123.4      125.1  123.6
5     6  123.3      124.8  123.7
6     7  123.3      125.1  123.7
7     8  123.6      125.0  123.6
8     9  123.5      124.8  123.6
9    10  123.5      124.7  123.8
10   11  123.5      124.5  123.7
11   12  123.7      125.2  123.8
12   13  123.6      124.4  123.5
13   14  123.7      124.7  123.7
14   15  123.9      124.9  123.0
15   16  124.0      124.5  123.8
16   17  124.2      124.8  123.8
17   18  123.9      124.8  123.1
18   19  123.8      124.5  123.7
19   20  123.8      124.6  123.7
20   21  124.0      125.0  123.8
21   22  123.9      124.7  123.8
22   23  123.6      124.9  123.7
23   24  123.5      124.9  123.8
24   25  123.4      124.7  123.7
25   26  123.4      124.2  123.8
26   27  123.4      124.7  123.8
27   28  123.4      124.8  123.8
28   29  123.3      124.4  123.9
29   30  123.3      124.6  123.8
30   31  123.5      124.4  123.9
31   32  123.6      124.0  123.9
32   33  123.8      124.7  123.9
33   34  123.9      124.4  123.9
34   35  123.9      124.6  123.6
35   36  123.8      124.6  123.2
36   37  123.9      124.6  123.1
37   38  123.7      124.8  123.0
38   39  123.8      124.3  122.9
39   40  123.8      124.0  123.0

In [55]:
# Kich thuoc mau
n = len(df6['Home'])

#Tính muy
muy = 120

# Tinh trung binh mau
tb = round(df6['Home'].mean(), 4)

# Tinh do lech chuan mau
dlc = 0.24

print('Kich thuoc mau: ', n)
print('Muy: ', muy)
print('Trung binh mau: ', tb)
print('Do lech chuan mau: ', dlc)

Kich thuoc mau:  40
Muy:  120
Trung binh mau:  123.6625
Do lech chuan mau:  0.24


In [56]:
#Tính miền bác bỏ
t_alpha_chia_2 = st.t.isf(0.01**2, df=n-1)
t_alpha_chia_2
print("Miền bác bỏ của giả thuyết là:{t:|t| >", t_alpha_chia_2,"}")

Miền bác bỏ của giả thuyết là:{t:|t| > 4.104681255441615 }


In [57]:
#Tính trị số kiểm định 
import math
t = (tb - muy) / (dlc / math.sqrt(n))
print("Trị số kiểm định: ",t)

Trị số kiểm định:  96.51534941972227


Vậy trị số kiểm định nằm trong vùng bác bỏ => bác bỏ giả thuyết "Mục tiêu cung cấp điện cho mỗi hộ gia đình là 120V"

## Kiểm định đặc trưng kỳ vọng ($\mu$) không biết ($\sigma$)

### Bài 7

Dataset 19(__SCREWS__) chứa thông tin khảo sát chiều dài một số đinh ốc được cung cấp bởi công ty Crown Bolt (đơn vị inches)

Trên bao bì của các đinh ốc do công ty sản xuất có ghi chiều dài của các đinh ốc là 3/4 inches. Sử dụng mẫu trên bạn hãy cho biết chiều dài của đinh ốc lấy từ mẫu thu nhận được từ dữ liệu trên có nhất quán với chiều dài ghi trên bao bì hay không? Kiểm định với mức ý nghĩa 5%

### Bài 8

Dataset 13 __(VOLTAGE)__ liệt kê các mức điện áp được đo tại các hộ gia đình bởi các máy đo khác nhau: tại nhà (HOME), tại máy phát độc lập (Generator), tại trạm phát (UPS).

Công ty cung cấp điện cho vùng này cam kết rằng: __"Mục tiêu cung cấp điện cho mỗi hộ gia đình là 120V"__ .

Với mức ý nghĩa 1%. Bạn hãy kiểm tra xem công ty có làm đúng cam kết hay không?

In [58]:
df8 = pd.read_excel('/content/drive/My Drive/Colab Notebooks/13_VOLTAGE.xls')
df8

Day   Home  Generator    UPS
0     1  123.8      124.8  123.1
1     2  123.9      124.3  123.1
2     3  123.9      125.2  123.6
3     4  123.3      124.5  123.6
4     5  123.4      125.1  123.6
5     6  123.3      124.8  123.7
6     7  123.3      125.1  123.7
7     8  123.6      125.0  123.6
8     9  123.5      124.8  123.6
9    10  123.5      124.7  123.8
10   11  123.5      124.5  123.7
11   12  123.7      125.2  123.8
12   13  123.6      124.4  123.5
13   14  123.7      124.7  123.7
14   15  123.9      124.9  123.0
15   16  124.0      124.5  123.8
16   17  124.2      124.8  123.8
17   18  123.9      124.8  123.1
18   19  123.8      124.5  123.7
19   20  123.8      124.6  123.7
20   21  124.0      125.0  123.8
21   22  123.9      124.7  123.8
22   23  123.6      124.9  123.7
23   24  123.5      124.9  123.8
24   25  123.4      124.7  123.7
25   26  123.4      124.2  123.8
26   27  123.4      124.7  123.8
27   28  123.4      124.8  123.8
28   29  123.3      124.4  123.9
29   30  123.3      124.6  123.8
30   31  123.5      124.4  123.9
31   32  123.6      124.0  123.9
32   33  123.8      124.7  123.9
33   34  123.9      124.4  123.9
34   35  123.9      124.6  123.6
35   36  123.8      124.6  123.2
36   37  123.9      124.6  123.1
37   38  123.7      124.8  123.0
38   39  123.8      124.3  122.9
39   40  123.8      124.0  123.0

In [59]:
# Kich thuoc mau
n = len(df8['Home'])

#Tính muy
muy = 120

# Tinh trung binh mau
tb = round(df8['Home'].mean(), 4)

# Tinh do lech chuan mau
dlc = round(df8['Home'].std(), 4)

print('Kich thuoc mau: ', n)
print('Muy: ', muy)
print('Trung binh mau: ', tb)
print('Do lech chuan mau: ', dlc)

Kich thuoc mau:  40
Muy:  120
Trung binh mau:  123.6625
Do lech chuan mau:  0.2404


In [60]:
#Tính miền bác bỏ
t_alpha_chia_2 = st.t.isf(0.01**2, df=n-1)
t_alpha_chia_2
print("Miền bác bỏ của giả thuyết là:{t:|t| >", t_alpha_chia_2,"}")

Miền bác bỏ của giả thuyết là:{t:|t| > 4.104681255441615 }


In [61]:
#Tính trị số kiểm định 
import math
t = (tb - muy) / (dlc / math.sqrt(n))
print("Trị số kiểm định: ",t)

Trị số kiểm định:  96.35475815612871


Vậy trị số kiểm định nằm trong vùng bác bỏ => bác bỏ giả thuyết "Mục tiêu cung cấp điện cho mỗi hộ gia đình là 120V"

### Bài 9

Dataset 2 __(BODYTEMP)__ chứa thông tin khảo sát nhiệt độ cơ thể từ một nhóm người vào cùng một thời điểm lúc 12h trưa.

Có một phát biểu nói rằng, nhiệt độ cơ thể người đo lúc 12h trưa trong quần thể này là 98,6°

Dựa vào mẫu trên, bạn hãy cho biết phát biểu trên đúng hay sai ở mức ý nghĩa là 5%.

In [62]:
df9 = pd.read_excel('/content/drive/My Drive/Colab Notebooks/02_BODYTEMP.xls')
df9

98.6
0    98.6
1    98.0
2    98.0
3    99.0
4    98.4
..    ...
100  97.8
101  98.4
102  97.4
103  98.0
104  97.0

[105 rows x 1 columns]

### Bài 10

__FICO Score__ là đại lượng dùng để đo khả năng tín dụng của một cá nhân hoặc một tập thể. Sử dụng FICO Score, ngân hàng có thể quyết định xem có nên cho khách hàng mượn tiền hay không?

Dataset 24 __(FICO)__ chứa thông tin điểm tín dụng (FICO Score) của một số cá nhân trong vùng.

Sử dụng dữ liệu mẫu từ, bạn hãy cho biết ý kiến: __"điểm tín dùng trung bình của quần thể là 678"__ là đúng hay sai với mức ý nghĩa 5%

In [63]:
df10 = pd.read_excel('/content/drive/My Drive/Colab Notebooks/24_FICO.xls')
df10

708
0   713
1   781
2   809
3   797
4   793
..  ...
94  796
95  611
96  709
97  697
98  732

[99 rows x 1 columns]

---